plugins/main/macvlan/macvlan.go

```diff
@@ -308,6 +308,12 @@ func cmdAdd(args *skel.CmdArgs) error {
 					_ = arping.GratuitousArpOverIface(ipc.Address.IP, *contVeth)
 				}
 			}
+            		// add a gateway route
+            		link, err := netlink.LinkByName(args.IfName)
+            		route := netlink.Route{LinkIndex: link.Attrs().Index, Gw:ipamResult.IPs[0].Gateway, Flags: 0x10}
```

Comment:
Be careful: this code can panic or return several errors:
1 - IPAM returning no addresses is pathological but valid
2 - IPAM results may not set a gateway.

`ipam.ConfigureInterface()` handles these cases correctly.

Reply:
@squeed you are right regarding the handling of this. Checks should be in place to make sure the response has IPs and a gateway.

ipam.ConfigureIface() is handling the gateway, the only problem is that after getting the v4gw and v6gw, it is only configuring the gateway if the IPAM response has "routes" configured as well. This is not the behavior we are seeing with other CNI plugins for e.g SRIOV CNI.

Based on the spec "gateway" under "IPs" indicate that it is the default gateway and "routes" should be used when you have to configure static routes where both the "dst" and "gw" towards a particular host or a subnet should be specified.

@syelburgi based on the comments seems like instead of adding a function to add the route here, you should fix ipam.ConfiugreIface() to properly handle route addition